arwen/host/execution.go

```diff
@@ -199,11 +199,55 @@ func copyTxHashesFromContext(copyEnabled bool, runtime arwen.RuntimeContext, inp
 
 // ExecuteOnDestContext pushes each context to the corresponding stack
 // and initializes new contexts for executing the contract call with the given input
-func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) (vmOutput *vmcommon.VMOutput, asyncInfo *arwen.AsyncContextInfo, gasUsedBeforeReset uint64, err error) {
+func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) (vmOutput *vmcommon.VMOutput, asyncInfo *arwen.AsyncContextInfo, err error) {
 	log.Trace("ExecuteOnDestContext", "caller", input.CallerAddr, "dest", input.RecipientAddr, "function", input.Function)
 
-	bigInt, _, metering, output, runtime, storage := host.GetContexts()
+	_, _, _, output, _, _ := host.GetContexts()
+
+	builtinFunctionExecuted := false
+	var builtinRevertInput *vmcommon.ContractCallInput
+
+	// TODO refactor/simplify this block
```

Comment:
let's refactor this. take the built in function execution into a separate.

Reply:
transferAndExecute model should be used

Reply:
Refactored.